native/libc.c

```diff
@@ -383,6 +386,30 @@ func_get_terminal_width(PyObject *self, PyObject *unused) {
   return PyLong_FromLong(w.ws_col);
 }
 
+static PyObject *
+func_wcswidth(PyObject *self, PyObject *args){
+    char *string;
+    if (!PyArg_ParseTuple(args, "s", &string)) {
+        return NULL;
+    }
+
+    const char *old_locale = setlocale(LC_CTYPE, "en_US.UTF-8");
```

Comment:
I'm pretty sure this does a file system operation, so the prompt length calculation will do more than one in general.

I guess the long term solution is to use our own library as in #366.

We might not be able to set the locale globally, because GNU readline is in the same process and also uses libc.  Stupid global variables...

Reply:
I just ran this locally and I got this wrong - setlocale returns the locale you _called_ it with, not the existing one. The way you get current locale is `setlocale(LC_CTYPE, NULL)`. Changing this real quick.

Reply:
And you're right, this mmaps /usr/lib/locale/locale-archive